exporters/provider_common/openshift.py

```diff
@@ -1,11 +1,61 @@
+import logging
+import re
+import time
 from datetime import datetime
-from typing import Union
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+from openshift.dynamic import DynamicClient, ResourceInstance
+from openshift.dynamic.exceptions import ResourceNotFoundError
+from openshift.dynamic.resource import ResourceField
 
 from pelorus.timeutil import parse_assuming_utc
 
 # https://docs.openshift.com/container-platform/4.10/rest_api/objects/index.html#io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta
 _DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
 
+SUPPORTED_REPLICA_OBJECTS = ["ReplicaSet", "ReplicationController"]
+
+# Cache threshold for each of the cached_parents_dict entries
+CACHE_THRESHOLD = 60 * 60 * 24
+cached_parents_dict: dict[str, ResourceInstance] = {}
+cached_parents_threshold: dict[str, float] = {}
+
+
+def _add_object_to_cache(uid: str, k8s_obj: ResourceInstance) -> None:
+    """
+    Create in-memory cache for the K8S objects, so we don't have
+    to query them each time.
+
+    We have also 'timeout' for each cache entry, which means
+    we won't grow the cache infinitely.
+    """
+    if uid not in cached_parents_threshold:
+        cached_parents_threshold[uid] = time.time()
+        cached_parents_dict[uid] = k8s_obj
+
+
+def _get_object_from_cache(uid: str) -> ResourceInstance:
+    """
+    Gets the object from the cache by it's uid.
+    """
+    return cached_parents_dict.get(uid)
+
+
+def _remove_expired_objects() -> None:
+    """
+    Cleanup function to remove expired objects from the cache.
+    """
+    current_time = time.time()
+    expired_keys = []
```

Comment:
[applies for others places]
Do not like creating a variable for storing the values of a for loop, we should use list comprehension 

example
```py
    expired_keys = [
        uid
        for uid, insertion_time in cached_parents.items()
        if current_time - insertion_time > CACHE_THRESHOLD
    ]
```

Reply:
fixed. what other places do you want to use list comprehension? I am fine using them if within the list comprehension there is no call to the external method as it's harder to debug later.